src/go/k8s/pkg/console/configmap.go

```diff
@@ -203,7 +203,14 @@ func (cm *ConfigMap) genLogin(ctx context.Context) (e EnterpriseLogin, err error
 		}
 		enterpriseLogin.JWTSecret = string(jwt)
 
-		switch { // nolint:gocritic // will support more providers
+		switch {
+		case provider.RedpandaCloud != nil:
+			enterpriseLogin.RedpandaCloud = &EnterpriseLoginRedpandaCloud{
+				Enabled:        provider.RedpandaCloud.Enabled,
+				Domain:         provider.RedpandaCloud.Domain,
+				Audience:       provider.RedpandaCloud.Audience,
+				AllowedOrigins: provider.RedpandaCloud.AllowedOrigins,
+			}
```

Comment:
Couldn't we reuse the same type? 

```suggestion
		case provider.RedpandaCloud != nil:
			enterpriseLogin.RedpandaCloud = provider.RedpandaCloud
```

This would require to leave only one implementation and remove 
https://github.com/redpanda-data/redpanda/blob/2307fd2a88f6912931bc4502090171291badbfb8/src/go/k8s/pkg/console/console.go#L77-L83

yaml tags would need to be added to the definition (I think).

Reply:
I didn't want to put yaml tags in the API definition because it might confuse users. But it's a minor thing, I updated them to use the API definition.